sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BatchLoads.java

```diff
@@ -113,6 +113,9 @@
   // If user triggering is supplied, we will trigger the file write after this many records are
   // written.
   static final int FILE_TRIGGERING_RECORD_COUNT = 500000;
+  // If user triggering is supplied, we will trigger the file write after this many bytes are
+  // written.
+  static final long FILE_TRIGGERING_BYTE_COUNT = 100 * (1L << 20); // 100MiB
```

Comment:
It looks like we already have a memory limit for writing of 20 parallel writers with 64mb buffers. Should we limit this triggering to be 64mbs as well so that it fits in one chunk?

CC: @reuvenlax Any suggestions here?

Reply:
@lukecwik Having the same limit as a buffer actually makes sense to me, but can you direct me towards where might I find that limit? I can see it in the comments for `DEFAULT_MAX_NUM_WRITERS_PER_BUNDLE`, but instead of hardcoding 64MB here as well, I would rather reference the original limit directly.

Reply:
The default comes from this constant:
https://www.javadoc.io/static/com.google.cloud.bigdataoss/util/1.9.17/com/google/cloud/hadoop/util/AsyncWriteChannelOptions.html#UPLOAD_CHUNK_SIZE_DEFAULT

The user can override the default using this pipeline option: https://github.com/apache/beam/blob/b2a6f46fb21709cc1927ce1950a38a922dce0a35/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/options/GcsOptions.java#L91

Reply:
done in [`bcd4ba9` (#22953)](https://github.com/apache/beam/pull/22953/commits/bcd4ba912a224aaba31be71b5eade04a2219de09)

Reply:
@lukecwik 
On second thought, I think there is a problem with using this 64MB default. We only flush the batch inside `GroupIntoBatches`, once the `storedBatchSizeBytes` is greater than or equal to the limit. So if we make the limit 64MB, more than likely we will flush just a bit more than 64MB so we won't fit into the 64MB buffer.

So either the triggering byte count should be x% smaller than the 64MB default, or `GroupIntoBatches` has to be modified that it the current element would make it go over the byte size limit, then fire the batch without that element being added to the it first. The second seems like a better solution, but I assume doing the `storedBatchSizeBytes.read()` sooner would have performance impact.

Reply:
Sorry, didn't see this comment but I agree that we should change GroupIntoBatches to ensure that if we add an element that would make it go over the limit we would flush a batch.

Pseudo-code would be like:
```
byteSize = measure(obj)
if (byteSize >= byteSizeLimit) {
  output obj as single element batch
  continue
}
if (byteSize + previousNumBytes > byteSizeLimit) {
  output all buffered elements as batch
} else {
  add obj to batch
}
```